### PR TITLE
Include filepost.pyi and util/url.pyi

### PR DESCRIPTION
**Pull request by savarin**
Addresses https://github.com/urllib3/urllib3/issues/1897.

This PR adds type stubs for `filepost.py` and `util/url.py`, adapted from the respective `.pyi` files from `typeshed`.

### CODE REVIEW

**Comment by savarin** [comment]
Error counts based on stub files:

```
nox > all errors count: 855
(venv) urllib3 $ touch src/urllib3/contrib/__init__.pyi
nox > all errors count: 855
(venv) urllib3 $ touch src/urllib3/filepost.pyi
nox > all errors count: 843
(venv) urllib3 $ touch src/urllib3/util/url.pyi
nox > all errors count: 778
```

**Comment by sethmlarson** [comment]
Thanks for this, I'll give this a review hopefully soon.
Btw is the other PR that's still open still necessary or is everything in that PR already merged?

**Comment by savarin** [comment]
> Thanks for this, I'll give this a review hopefully soon.
> Btw is the other PR that's still open still necessary or is everything in that PR already merged?

Thank you! I looked into adding `_collections.pyi` and `request.pyi` but will do that in another PR instead. Re: other PR, good catch - the content's already merged so just closed it.

File: src/urllib3/filepost.pyi [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from typing import Any, Generator, List, Mapping, Optional, Tuple, Union
+
+from . import fields
+
+Fields = Union[Mapping[str, str], List[Tuple[str]]]
+Iterator = Generator[Tuple[str], None, None]
+
+RequestField = fields.RequestField
```
Comment (pquentin) [inline src/urllib3/filepost.pyi:8]: `RequestField` is unused, but it looks like `iter_field_objects` and `encode_multipart_formdata` should accept this type too?

Comment (savarin) [reply src/urllib3/filepost.pyi:5]: @pquentin Good catch, amended.

**Review by pquentin — APPROVED** [review]
Thank you!

File: src/urllib3/filepost.pyi [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from . import fields
+
+RequestField = fields.RequestField
+Fields = Union[Mapping[str, str], List[Tuple[str]], List[RequestField]]
+Iterator = Generator[Tuple[str], None, None]
+
+writer: Any
+
+def choose_boundary() -> str: ...
+def iter_field_objects(fields: Fields) -> Iterator: ...
```
Comment (sethmlarson) [inline src/urllib3/filepost.pyi:12]: Typically Iterator should also contain the type of the item returned from the iterator. `Iterator[X]`

File: src/urllib3/filepost.pyi [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+def iter_fields(fields: Fields) -> Iterator: ...
+def encode_multipart_formdata(
+    fields: Fields, boundary: Optional[str]
+) -> Tuple[str]: ...
```
Comment (sethmlarson) [inline src/urllib3/filepost.pyi:16]: This return type doesn't look right `Tuple[str]` indicates a one-tuple, need to be Tuple[bytes, str]?

File: src/urllib3/util/url.pyi [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+url_attrs: List[str]
+
+class Url:
+    slots: Any
+    def __new__(
+        cls,
+        scheme: Optional[str],
+        auth: Optional[str],
+        host: Optional[str],
+        port: Optional[str],
```
Comment (sethmlarson) [inline src/urllib3/util/url.pyi:16]: Port is an optional int

File: src/urllib3/util/url.pyi [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @property
+    def request_uri(self) -> str: ...
+    @property
+    def netloc(self) -> str: ...
+    @property
+    def url(self) -> str: ...
+
+def split_first(s: str, delims: str) -> Tuple[str, str, Optional[str]]: ...
+def parse_url(url: str) -> Url: ...
+def get_host(url: str) -> Union[str, Tuple[str]]: ...
```
Comment (sethmlarson) [inline src/urllib3/util/url.pyi:32]: Return type should be Tuple[str, str, Optional[int]]

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Sorry didn't get to reviewing this earlier.

I think there is a problem with mypy not comparing the .py files to our .pyi files because these issues weren't caught:

**Pull request merged**